src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -183,13 +209,42 @@ public static PhysicalAddress Parse(string address)
                 validCount++;
             }
 
-            // we too few characters after the last dash
-            if (validCount < 2)
+            // we too few characters after the last delimiter
+            if (validCount < maxSegmentLength)
             {
                 throw new FormatException(SR.Format(SR.net_bad_mac_address, address));
             }
 
             return new PhysicalAddress(buffer);
         }
+
+        private static int GetMaxSegmentLength(string address, char? delimiter)
+        {
+            int segments = 1;
+            int maxSegmentLength = 0;
+            for (int i = 0; i < address.Length; i++)
+            {
+                if (address[i] == delimiter)
+                {
+                    if (maxSegmentLength == 0)
+                    {
+                        maxSegmentLength = i;
+                    }
+                    //segments - 1 = num of delimeters -> check current segment length
```

Comment:
Nit: please move the comment into the else if block... we generally prefer not to have comments between cascading if/else ifs.

Also, can you elaborate on the comment and specifically what it's checking?  It looks like this is validating that every segment is the same length, but I'm not understanding why it's using %.  Initially I thought it was because you were allowing for missing delimiters, but then the check later seems to obviate that.  If we only support entries where segments are either 0 length or else all segments are equal length, shouldn't this be checking simply that the length of this segment matches maxSegmentLength?

Reply:
I updated the comment a bit, but basically every time I hit a delimiter, I'm checking that the "non delimiter length" is a multiple of the valid segment length. Effectively, this is checking the length of the segment that was most recently iterated over (a previous segment would have already thrown, so if the most recent segment is the same length as the validSegmentLength, it's a multiple of it).

I do this because I'm not explicitly keeping track of the current segment length via a separate value. I can do that if that more sense to you, but that was my thinking.